.github/workflows/release.yml

```diff
@@ -0,0 +1,304 @@
+      #       exit 1
+      #     fi
+      #     echo "Tag is from main branch (commit: $TAG_COMMIT)"
+      #
+      # # Make sure that Cargo.toml version for each tool is the same as the tag
+      # - name: Verify version consistency
+      #   shell: bash
+      #   run: |
+      #     # Extract version from tag
+      #     TAG_VERSION="${GITHUB_REF#refs/tags/v}"
+      #
+      #     # Check each tool's Cargo.toml version
+      #     for tool in move-mutation-test move-mutator move-spec-test; do
+      #       CARGO_VERSION=$(grep "^version" $tool/Cargo.toml | head -1 | cut -d'"' -f2)
+      #       if [ "$CARGO_VERSION" != "$TAG_VERSION" ]; then
+      #         echo "Error: Version mismatch for $tool!"
+      #         echo "   Tag version: $TAG_VERSION"
+      #         echo "   Cargo.toml version: $CARGO_VERSION"
+      #         exit 1
+      #       fi
+      #       echo "$tool version matches: $CARGO_VERSION"
+      #     done
+
+      - name: Enable long paths on Windows
+        if: runner.os == 'Windows'
+        run: |
+          git config --system core.longpaths true
+          reg add HKLM\SYSTEM\CurrentControlSet\Control\FileSystem /v LongPathsEnabled /t REG_DWORD /d 1 /f
+        shell: pwsh
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install Aptos CLI
+        shell: bash
+        run: |
+          if [ "${{ runner.os }}" = "macOS" ]; then
+            echo "Installing Aptos CLI via Homebrew..."
+            brew install aptos
+          elif [ "${{ runner.os }}" = "Windows" ]; then
+            echo "Installing Aptos CLI via Chocolatey..."
+            choco install aptos -y
+          else
+            echo "Installing Aptos CLI via installer script..."
+            curl -fsSL "https://aptos.dev/scripts/install_cli.sh" | sh
+            source ~/.profile
+            APTOS_PATH=$(dirname $(which aptos))
+            echo "$APTOS_PATH" >> $GITHUB_PATH
+          fi
+          aptos --version
+
+      - name: Cache Cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-${{ matrix.target }}-cargo-release-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build all tools
+        shell: bash
+        run: |
+          cargo build --release --bin move-mutation-test
+          cargo build --release --bin move-mutator
+          cargo build --release --bin move-spec-test
+
+      - name: Basic smoke test
+        shell: bash
+        run: |
+          ./target/release/move-mutation-test${{ matrix.binary_ext }} --version
+          ./target/release/move-mutator${{ matrix.binary_ext }} --version
+          ./target/release/move-spec-test${{ matrix.binary_ext }} --version
+
+      - name: Test move-mutation-test tool on simple_move_2_features (Linux, Mac, Windows)
```

Comment:
We're not testing the move-spec-test tool because I couldn't successfully install the dependencies for it (copy pasted from the exsting workflow related to it).
The move-mutator I'm not sure how to test an if it's actually relevant.